### PR TITLE
[v2.8] Fix etcd version not displaying version in CheckETCDVersion

### PR DESCRIPTION
**Pull request by markusewalker**
## Problem

For RKE1 custom cluster provisioning, we perform a check for the etcd etcdctl version in each etcd node. This test specifically was created when we are doing release checks. This is not working due to an issue seen in the function `CheckETCDVersion`.
 
## Solution

Fixed the issue in `CheckETCDVersion` function.
 
### Manual Testing

Manually ran the RKE1 custom cluster test and this was successful.

### Automated Testing
Jenkins jobs will be provided offline to reviewers.

### CODE REVIEW

File: tests/v2/validation/provisioning/permutations/permutations.go
Comment (slickwarren) [inline tests/v2/validation/provisioning/permutations/permutations.go:133]: so, there's no version checking at all now? What is this test doing if it doesn't check the version?

Comment (slickwarren) [reply tests/v2/validation/provisioning/permutations/permutations.go:170]: logging the output isn't checking the output for correct version or anything though. We should still check if the version is valid. Maybe instead of checking if it is not empty, we should just find out what is expected and base the test on that.

Comment (markusewalker) [reply tests/v2/validation/provisioning/permutations/permutations.go:170]: Let's talk offline about this, there's a bit confusion here I think.

Comment (markusewalker) [reply tests/v2/validation/provisioning/permutations/permutations.go]: Fixed this, running new jobs offline.

File: tests/framework/extensions/clusters/clusters.go
Comment (slickwarren) [inline tests/framework/extensions/clusters/clusters.go:326]: was this meant to be part of this pr?

File: tests/framework/extensions/rke1/componentchecks/etcdversion.go
Comment (slickwarren) [inline tests/framework/extensions/rke1/componentchecks/etcdversion.go:27]: will this work on multi-node clusters?

Comment (markusewalker) [reply tests/framework/extensions/rke1/componentchecks/etcdversion.go:27]: I will check. We only need to run this on one etcd node, hence the break. But I will verify that and follow back.

File: tests/framework/extensions/clusters/clusters.go
Comment (markusewalker) [reply tests/framework/extensions/clusters/clusters.go:326]: Yeah, I wrote offline about it. This is a small change that was missed in separate etcd snapshot testing for S3 clusters.

Comment (markusewalker) [reply tests/framework/extensions/clusters/clusters.go:326]: Resolved offline.

File: tests/framework/extensions/rke1/componentchecks/etcdversion.go
Comment (markusewalker) [reply tests/framework/extensions/rke1/componentchecks/etcdversion.go]: Resolved, running new job.

**Review by slickwarren — COMMENTED** [review]
approved, but I think the extensions changes will have to get approval from Israel before merging / will end up being shepherd changes.

**Review by slickwarren — APPROVED** [review]

**Pull request closed without merging**
